test/parallel/test-buffer-regression-649.js

```diff
@@ -6,8 +6,14 @@ const SlowBuffer = require('buffer').SlowBuffer;
 
 // Regression test for https://github.com/nodejs/node/issues/649.
 const len = 1422561062959;
-assert.throws(() => Buffer(len).toString('utf8'));
-assert.throws(() => SlowBuffer(len).toString('utf8'));
-assert.throws(() => Buffer.alloc(len).toString('utf8'));
-assert.throws(() => Buffer.allocUnsafe(len).toString('utf8'));
-assert.throws(() => Buffer.allocUnsafeSlow(len).toString('utf8'));
+const lenLimitMsg = /^RangeError: Invalid typed array length$/;
+assert.throws(() => Buffer(len).toString('utf8'),
+              lenLimitMsg);
```

Comment:
It's really a personal preference. Sometimes with large PRs it's easier for the reviewer(s) if the entire thing doesn't have to be re-reviewed after each amended change. It can be tricky though if a PR contains multiple logically divided commits and you want to amend the correct commit. For something of this size, I don't think it matters much.